traffic_ops/traffic_ops_golang/asn/asns.go

```diff
@@ -47,26 +47,29 @@ func GetRefType() *TOASN {
 }
 
 //Implementation of the Identifier, Validator interface functions
-func (asn *TOASN) GetID() (int, bool) {
+func (asn TOASN) GetID() (int, bool) {
 	return asn.ID, true
 }
 
-func (asn *TOASN) GetAuditName() string {
+func (asn TOASN) GetAuditName() string {
 	return strconv.Itoa(asn.ASN)
 }
 
-func (asn *TOASN) GetType() string {
+func (asn TOASN) GetType() string {
 	return "asn"
 }
 
 func (asn *TOASN) SetID(i int) {
 	asn.ID = i
 }
 
-func (asn *TOASN) Validate(db *sqlx.DB) []error {
+func (asn TOASN) Validate(db *sqlx.DB) []error {
 	errs := []error{}
-	if asn.ASN < 1 {
-		errs = append(errs, errors.New(`ASN 'asn' is required.`))
+	if asn.ASN < 0 {
```

Comment:
Should this be the Validation framework doing the checks?

Reply:
since this was only the simple comparison,  I decided against that.   This isn't likely to ever have any other checks.

Reply:
@dew after changing to the nullable struct,  made much more sense to do this -- changed to use validation framework